CentralUnit/Communication/CommunicationFailureEventArgs.cs

```diff
@@ -0,0 +1,57 @@
+﻿// <copyright file="CommunicationFailureEventArgs.cs" company="Moto Gymkhana">
+//     Copyright (c) Moto Gymkhana. All rights reserved.
+// </copyright>
+namespace Communication
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+
+    /// <summary>
+    /// Type of communication failure.
+    /// </summary>
+    public enum FailureType
+    {
+        /// <summary>
+        /// Communication channel got disconnected
+        /// </summary>
+        Disconnect,
+        
+        /// <summary>
+        /// Communication timeout occurred.
+        /// </summary>
+        Timeout,
+
+        /// <summary>
+        /// Generic failure occurred.
+        /// </summary>
+        Generic
+    }
+    
+    /// <summary>
+    /// Event args to pass data about a communication failure with a Communication Failure event.
+    /// </summary>
+    public class CommunicationFailureEventArgs : EventArgs
```

Comment:
Would it make sense for this class to take an exception as well? i can imagine this will be thrown in response to an exception

Reply:
It doesn't take exceptions directly, but string representations of exceptions.